crates/runtime/src/lib.rs

```diff
@@ -55,6 +67,7 @@ pub struct Builder {
     engine: Engine,
     module: Module,
     state_builder: SlightCtxBuilder,
+    io_redirects: Option<IORedirects>,
```

Comment:
I don't know if this could simplify things, but if you define `IORedirects` as a struct of `Option<PathBuf>` for each std-io, this will give granularity to users to specify only one of the IO path. Then you can say `io_redirects` is `IORedirects` instead of `Option<IORedirects>` and it could be defaulted to a struct with Nones.

Reply:
I think I'd like to keep the optional on IORedirects as it really is optional for args. However, I did make the paths within the struct as optional as well. This provides the ability to specify some, but not all.